include/sys/zfs_sa.h

```diff
@@ -87,6 +88,8 @@ typedef enum zpl_attr {
 #define	SA_UID_OFFSET		24
 #define	SA_GID_OFFSET		32
 #define	SA_PARENT_OFFSET	40
+#define	SA_FLAGS_OFFSET		48
+#define	SA_PROJID_OFFSET	128
```

Comment:
It would be a good idea to keep all the mandatory attrs together (i.e. MODE through LINKS).

If you're asking about zfs_sa_upgrade(), I don't see how it matters if you switch the order of RDEV and PROJID.